src/platform.ts

```diff
@@ -140,19 +139,22 @@ export function getAvailablePowerShellExes(platformDetails: IPlatformDetails): I
             });
         }
 
+        const psCoreInstallPath =
+            (!platformDetails.isProcess64Bit ? process.env.ProgramW6432 : process.env.ProgramFiles) + "\\PowerShell";
+
         if (fs.existsSync(psCoreInstallPath)) {
+            const arch = platformDetails.isProcess64Bit ? "(x64)" : "(x86)";
             const psCorePaths =
                 fs.readdirSync(psCoreInstallPath)
                 .map((item) => path.join(psCoreInstallPath, item))
-                .filter((item) => fs.lstatSync(item).isDirectory())
+                .filter((item) => {
+                    const exePath = path.join(item, "pwsh.exe");
+                    return fs.lstatSync(item).isDirectory() && fs.existsSync(exePath);
+                })
                 .map((item) => {
-                    let exePath = path.join(item, "pwsh.exe");
-                    if (!fs.existsSync(exePath)) {
-                        exePath = path.join(item, "powershell.exe");
-                    }
-
+                    const exePath = path.join(item, "pwsh.exe");
                     return {
-                        versionName: `PowerShell Core ${path.parse(item).base}`,
+                        versionName: `PowerShell Core ${path.parse(item).base} ${arch}`,
```

Comment:
`exePath: path.join(item, "pwsh.exe"),` below this avoids a variable and still gets readability here, provided I haven't missed anything

Reply:
In fact it could be:

`.map((item) => {
    versionName: `PowerShell Core ${path.parse(item),base} ${arch}`,
    exePath: path.join(item, "pwsh.exe")
});

Reply:
Yes, that is cleaner! Thanks.  In removing that `fs.existsSync()` check, I should have seen the reduced form.

Reply:
It was a little trickier than that but not by much.

Reply:
Ah sorry, I was suspicious about my knowledge of TypeScript's lambda expression rules.